lib/puppet/util/metric.rb

```diff
@@ -4,17 +4,24 @@
 
 # A class for handling metrics.  This is currently ridiculously hackish.
 class Puppet::Util::Metric
+  include Puppet::Util::PsychSupport
   include Puppet::Network::FormatSupport
 
   attr_accessor :type, :name, :value, :label
   attr_writer :values
 
   def self.from_data_hash(data)
-    metric = new(data['name'], data['label'])
-    metric.values = data['values']
+    metric = allocate
+    metric.initialize_from_hash(data)
     metric
   end
 
+  def initialize_from_hash(data)
+    @name = data['name']
+    @label = data['label'] || self.class.labelize(@name)
```

Comment:
Can `initialize_from_hash` and `initialize` be collapsed? Right now `initialize` does a `to_s` on the name, but `from_data_hash`/`initialize_from_hash` doesn't.